packages/cli/index.ts

```diff
@@ -119,3 +136,15 @@ export const crash = (msg?: string): never => {
 	}
 	exit(1);
 };
+
+export const error = (msg?: string, extra?: string): never => {
```

Comment:
what's the difference between this and `crash` above? couldn't `crash` be generalized instead of defining a new function?

Reply:
It prints an "ERROR" prefix, and I feel like an error method can convey different messaging from crash

Reply:
I see, regarding the prefix again it could be set conditionally (although I would be tempted to just always include it)

___

I see your point in `crash` vs `error`, but I must say that I don't fully agree, `crash` is just a synonym for "unrecoverable error" isn't it?

Also it does to me look like `crash` is just used for displaying errors (+ exiting the process):
![Screenshot 2023-11-30 at 13 07 33](https://github.com/cloudflare/workers-sdk/assets/61631103/436a6a04-ab0a-464e-a60b-5950b594609d)

___

If we for example take:
```ts
crash("Failed to create pages project. See output above.");
```
([source](https://github.com/cloudflare/workers-sdk/blob/f20b02a90cea851c8b8d050cfc34ec310e3ff8f0/packages/create-cloudflare/src/pages.ts#L222))
could you explain to me the difference between that and
```ts
error(
			"There has been an error while loading your deployments: \n " +
				err.message
		);
```
([source](https://github.com/cloudflare/workers-sdk/pull/4310/files#diff-4e6941af0591cf1d2252f8793c29c620c19f9f35548769f1ce39b53226cd347eR79-R82))
?

They seem to convey practically the same exact meaning/semantic to me (and I don't see the benefit in having one being called `error` and the other `crash`)

___

Another example:
```ts
crash("Pages project isn't ready yet. Please try deploying again later.");
```
([source](https://github.com/cloudflare/workers-sdk/blob/f20b02a90cea851c8b8d050cfc34ec310e3ff8f0/packages/create-cloudflare/src/pages.ts#L251))
vs
```ts
	error(
		"Couldn't retrieve a new deployment: " + getDeploymentError.message
	);
```
([source](https://github.com/cloudflare/workers-sdk/pull/4310/files#diff-47cec4445ac6c9fb7fba1c8cb8f273d48d999ad3e7928176a71bfcea5a6080d4R313-R315))

Both indicate that the result from some kind of deployment isn't retrievable.

could you explain why the first is a crash and the second is an error?

Reply:
Well, for us we just found it to differentiate the rendering style. We didn't find changing the crash function signature to expose an error prefix an appealing thing to do at first.